config/routes.rb

```diff
@@ -2,6 +2,9 @@
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
   root 'users#index'
   resources :users, only: [:index, :show] do
-    resources :posts, only: [:index, :show]
+    resources :posts, only: [:index, :show, :new, :create] do
+      resources :comments, only: [:new, :create]
+      resources :likes, only: [:new, :create]
```

Comment:
* ## OPTIONAL 
* I strongly suggest you only make routes needed available as you have new routes for likes but there is no new action in your likes controller